test/integration/custom.test.js

```diff
@@ -0,0 +1,259 @@
+      case 'bulkInsert':
+        return this.formatInsert(spell);
+      case 'select':
+        return this.formatSelect(spell);
+      case 'update':
+        return this.formatUpdate(spell);
+      case 'delete':
+        return this.formatDelete(spell);
+      case 'upsert':
+        return this.formatUpsert(spell);
+      default:
+        throw new Error(`Unsupported SQL command ${spell.command}`);
+    }
+  }
+
+  formatUpdate(spell) {
+    const { Model, sets, whereConditions } = spell;
+    const { shardingKey } = Model;
+    const { escapeId } = Model.driver;
+    if (shardingKey) {
+      if (sets.hasOwnProperty(shardingKey) && sets[shardingKey] == null) {
+        throw new Error(`Sharding key ${Model.table}.${shardingKey} cannot be NULL`);
+      }
+      if (!whereConditions.some(condition => findExpr(condition, { type: 'id', value: shardingKey }))) {
+        throw new Error(`Sharding key ${Model.table}.${shardingKey} is required.`);
+      }
+    }
+
+    if (Object.keys(sets).length === 0) {
+      throw new Error('Unable to update with empty set');
+    }
+
+    const table = escapeId(spell.table.value);
+    const values = {};
+    Object.keys(spell.sets).reduce((obj, key) => {
+      obj[escapeId(Model.unalias(key))] = spell.sets[key];
+      return obj;
+    }, values);
+  
+    let whereArgs = [];
+    let whereClause = '';
+    if (whereConditions.length > 0) {
+      for (const condition of whereConditions) collectLiteral(spell, condition, whereArgs);
+      whereClause += `WHERE ${formatConditions(spell, whereConditions)}`;
+    }
+    return {
+      table,
+      whereArgs,
+      whereClause,
+      values,
+    };
+  }
+
+  formatDelete(spell) {
+    const { Model, whereConditions } = spell;
+    const { escapeId } = Model.driver;
+    const table = escapeId(spell.table.value);
+    let whereArgs = [];
+    let whereClause = '';
+    if (whereConditions.length > 0) {
+      for (const condition of whereConditions) collectLiteral(spell, condition, whereArgs);
+      whereClause += `WHERE ${formatConditions(spell, whereConditions)}`;
+    }
+    return {
+      table,
+      whereArgs,
+      whereClause,
+    };
+  }
+
+  formatMyInsert(spell) {
+    const { Model, sets } = spell;
+    const { escapeId } = Model.driver;
+    const table = escapeId(spell.table.value);
+    let values = {};
+
+    const { shardingKey } = Model;
+    if (shardingKey && sets[shardingKey] == null) {
+      throw new Error(`Sharding key ${Model.table}.${shardingKey} cannot be NULL.`);
+    }
+    for (const name in sets) {
+      const value = sets[name];
+      values[escapeId(Model.unalias(name))] = value instanceof Raw? SqlString.raw(value.value) : value;
+    }
+
+    return {
+      table,
+      values,
+    };
+  }
+
+};
+
+class CustomDriver extends SQLiteDriver {
+  static Spellbook = MySpellBook;
+
+  get driverName () {
+    return 'myCustomDriver';
+  }
+
+  async queryWithSpell(spell) {
```

Comment:
Overide `queryWithSpell` to use custom driver's query API

Reply:
`querySpell(spell)` shall suffice